tests/system/src/test/java/test/com/sun/javafx/text/TextLayoutTest.java

```diff
@@ -0,0 +1,430 @@
+            return font;
+        }
+        @Override public RectBounds getBounds() {
+            return null;
+        }
+    }
+
+    private void setContent(PrismTextLayout layout, Object... content) {
+        int count = content.length / 2;
+        TextSpan[] spans = new TextSpan[count];
+        int i = 0;
+        while (i < content.length) {
+            spans[i>>1] = new TestSpan(content[i++], content[i++]);
+        }
+        layout.setContent(spans);
+    }
+
+    private void assertLineCount(int lineCount) {
+        assertEquals(lineCount, layout.getLines().length, "lineCount");
+    }
+
+    private void assertLineBounds(RectBounds... rectBounds) {
+        assertLineCount(rectBounds.length);
+
+        TextLine[] lines = layout.getLines();
+
+        for (int i = 0; i < lines.length; i++) {
+            assertEquals(rectBounds[i], lines[i].getBounds(), "line " + i);
+        }
+    }
+
+    private void assertLineLocations(Point2D... locations) {
+        assertLineCount(locations.length);
+
+        TextLine[] lines = layout.getLines();
+
+        for (int i = 0; i < lines.length; i++) {
+            assertEquals(locations[i], lines[i].getRuns()[0].getLocation(), "line " + i);
+        }
+    }
+
+    private void assertGlyphsPerRun(int... glyphCount) {
+        GlyphList[] runs = layout.getRuns();
+
+        assertEquals(glyphCount.length, runs.length, "number of glyph counts given does not match number of runs");
+
+        for (int i = 0; i < runs.length; i++) {
+            assertEquals(glyphCount[i], runs[i].getGlyphCount(), "run " + i);
+        }
+    }
+
+    private void verifyLayout(int lineCount, int runCount, int... glyphCount) {
+        TextLine[] lines = layout.getLines();
+        assertEquals(lineCount, lines.length, "lineCount");
+        GlyphList[] runs = layout.getRuns();
+        assertEquals(runCount, runs.length, "runCount");
+        assertEquals(runCount, glyphCount.length, "runCount");
+        for (int i = 0; i < runs.length; i++) {
+            assertEquals(glyphCount[i], runs[i].getGlyphCount(), "run " + i);
+        }
+    }
+
+    private void verifyComplex(boolean... complex) {
+        GlyphList[] runs = layout.getRuns();
+        for (int i = 0; i < runs.length; i++) {
+            assertEquals(complex[i], runs[i].isComplex(), "run " + i);
+        }
+    }
+
+    /**
+     * These tests were broken for a long time (as early as 2013-06-28, see JDK-8087615).
+     *
+     * The reason they break is two fold:
+     *
+     * - Content that is split into multiple runs will have all runs set to "complex"
+     *   if at least one run is set to "complex", while this test was expecting the
+     *   runs containing non-complex characters to not become "complex".
+     *
+     * - The Tahoma font when used with a Thai character generates 2 glyphs per Thai
+     *   character, while this test was expecting 1 glyph.
+     */
+    @Disabled
+    @Test
+    void complexTestsThatAreBrokenSince2013() {
```

Comment:
@kevinrushforth
re: ancient tests

this is the only skipped test - quite possibly due to wrong expectations (about Thai)

Due to relative complexity of the change, I think we need all the tests we could get in this PR

Reply:
I added the ticket number in the `@Disabled`.  The ticket however is pretty useless to keep open I think.